pkg/sql/crdb_internal.go

```diff
@@ -8034,3 +8035,48 @@ CREATE TABLE crdb_internal.node_tenant_capabilities_cache (
 		return nil
 	},
 }
+
+func populateFromQuery(query string, op string) populateFunc {
+	return func(ctx context.Context, p *planner, _ catalog.DatabaseDescriptor, addRow func(...tree.Datum) error) (retErr error) {
+		rows, err := p.ExecCfg().InternalDB.Executor().QueryIteratorEx(
+			ctx,
+			op,
+			nil,
+			sessiondata.RootUserSessionDataOverride,
+			query,
+		)
+		if err != nil {
+			return err
+		}
+		defer func() {
+			retErr = errors.CombineErrors(retErr, rows.Close())
+		}()
+		for {
+			ok, err := rows.Next(ctx)
+			if err != nil {
+				return err
+			}
+			if !ok {
+				break
+			}
+			err = addRow(rows.Cur()...)
+			if err != nil {
+				return err
+			}
+		}
+		return nil
+	}
+}
+
+var crdbInternalSystemPrivileges = virtualSchemaTable{
```

Comment:
I think (but I'm not sure) that you could define this as a virtualSchemaView and this way you would not need to introduce `populateFromQuery` at all!

I investigated this a bit. In `master` we already have `crdb_internal.transaction_statistics_persisted` which is the same thing for `system.transaction_statistics`; I set up a logictest in which `testuser` was non-admin and querying the latter would correctly raise an error however it totally can query the vtable.

Reply:
Thanks I was looking for something like this, but did not know what to search for.